sdks/python/apache_beam/pipeline.py

```diff
@@ -1416,10 +1418,15 @@ def get_or_assign(self, obj=None, obj_type=None, label=None):
 
     return self._obj_to_id[obj]
 
+  def _normalize(self, str_value):
+    str_value = unicodedata.normalize('NFC', str_value)
+    return re.sub(r'[^a-zA-Z0-9-_]+', '-', str_value)
```

Comment:
TIL about [unicode normalization](https://en.wikipedia.org/wiki/Unicode_equivalence#Normalization). Just for my own edification, do you know why NFC is the right option here?

Reply:
I'm not 100% sure what exactly to use here. NFC and NFD seems to be the two main options and NFC seems to be the shorter of the two so I chose that: https://docs.python.org/3.6/library/unicodedata.html#unicodedata.normalize
Also, this will only apply to non-ASCII characters IIUC.

Reply:
Makes sense, thanks